app/src/main/java/com/cornellappdev/uplift/data/clients/ApolloReportClient.kt

```diff
@@ -0,0 +1,49 @@
+package com.cornellappdev.uplift.data.clients
+
+import android.util.Log
+import com.apollographql.apollo3.ApolloClient
+import com.cornellappdev.uplift.CreateReportMutation
+import com.cornellappdev.uplift.domain.clients.ReportClient
+
+/**
+ * A client for interacting with the report API.
+ * @param apolloClient The Apollo client to use for making requests.
+ */
+class ApolloReportClient(
+    private val apolloClient: ApolloClient
+) : ReportClient {
+    /**
+     * Creates a report.
+     * @param createdAt The time the report was created.
+     * @param description The description of the report.
+     * @param gymId The ID of the gym the report is about.
+     * @param issue The issue the report is about.
+     * @param userId The ID of the user creating the report.
+     * @return The created report.
+     */
```

Comment:
Damn dude, love the spec! Are you using AI complete to write this?? Super thorough

Reply:
Yeah mostly AI just some minor edits, github copilot is really nice!